src/model_signing/_cli.py

```diff
@@ -0,0 +1,451 @@
+
+    _supported_modes = ["sigstore", "key", "certificate"]
+
+    def get_command(
+        self, ctx: click.Context, cmd_name: str
+    ) -> Optional[click.Command]:
+        """Retrieves a command with a given name.
+
+        We use this to make Sigstore signing be the default, if it is missing.
+        """
+        if cmd_name in self._supported_modes:
+            return super().get_command(ctx, cmd_name)
+        return super().get_command(ctx, "sigstore")
+
+    def resolve_command(
+        self, ctx: click.Context, args: list[str]
+    ) -> tuple[Optional[str], Optional[click.Command], list[str]]:
+        """Resolves a command and its arguments.
+
+        We use this to make Sigstore signing be the default and correctly alter
+        the arguments. We are guaranteed that `args` has at least one element
+        (otherwise the help menu would be printed). This argument should be the
+        subcommand and would be removed as a result of this function, in
+        general.
+
+        However, if the first argument does not resolve to a supported PKI
+        method, then we inject "sigstore" as the subcommand (in `get_command`).
+        All that is left to do is to pass all `args` to the subcommand, without
+        removing anything.
+        """
+        if args[0] in self._supported_modes:
+            return super().resolve_command(ctx, args)
+        _, cmd, _ = super().resolve_command(ctx, args)
+        return cmd.name, cmd, args
+
+
+@click.group(
+    context_settings=dict(help_option_names=["-h", "--help"]),
+    epilog=(
+        "Check https://sigstore.github.io/model-transparency for "
+        "documentation and more details."
+    ),
+)
+@click.version_option(model_signing.__version__, "--version")
+def main() -> None:
+    """ML model signing and verification.
+
+    Use each subcommand's `--help` option for details on each mode.
+    """
+
+
+@main.group(name="sign", subcommand_metavar="PKI_METHOD", cls=PKICmdGroup)
+def _sign() -> None:
+    """Sign models.
+
+    Produces a cryptographic signature (in the form of a Sigstore bundle) for a
+    model. We support any model format, either as a single file or as a
+    directory.
+
+    We support multiple PKI methods, specified as subcommands. By default, the
+    signature is generated via Sigstore (as if invoking `sigstore` subcommand).
+
+    Use each subcommand's `--help` option for details on each mode.
+    """
+
+
+@_sign.command(name="sigstore")
+@_model_path_argument
+@_write_signature_option
+@click.option(
+    "--use_ambient_credentials",
+    type=bool,
+    is_flag=True,
+    help="Use credentials from ambient environment.",
+)
+@click.option(
+    "--identity_token",
+    type=str,
+    metavar="TOKEN",
+    help=(
+        "Fixed OIDC identity token to use instead of obtaining credentials "
+        "from OIDC flow or from the environment."
+    ),
+)
+@click.option(
+    "--use_staging",
```

Comment:
this was `--staging` in the CLI before. was this intentional to line up with the underlying API?

Reply:
Yeah, I intended the change to make it look like a desired action.